monkey/monkey_island/cc/services/config.py

```diff
@@ -467,3 +468,95 @@ def _format_pbas_from_flat_config(config: Dict):
         config.pop(flat_linux_filename_field, None)
         config.pop(flat_windows_command_field, None)
         config.pop(flat_windows_filename_field, None)
+
+    @staticmethod
+    def _format_propagation_from_flat_config(config: Dict):
+        formatted_propagation_config = {"network_scan": {}, "targets": {}}
```

Comment:
Maybe we should extract this into a dataclass in `common`? Then there would be no guessing what the `options` are. Also, a typo would be less likely

Reply:
Yeah, but probably not in this PR. I've added a task to #1597 to cover it.